### PR TITLE
Fixed Incorrect Types And Added Optional ClassName Props

### PR DESCRIPTION
**Pull request by zikiro**
- The types were incorrect and caused the code not to work without ts ignore in typescript.
- Added some additional props to make styling around the numbers more viable.

### CODE REVIEW

**Comment by zikiro** [comment]
@bluebill1049 Is this able to be merged or is there something additional needed?

File: src/flipNumbers.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import FlipNumber from './flipNumber';
 type Props = {
   numbers: string | Array<string>,
   nonNumberStyle?: { [string]: string | number },
```
Comment (bluebill1049) [inline src/flipNumbers.js]: could you fix this one as well?

**Review by bluebill1049 — COMMENTED** [review]
shall we only have a `className` prop instead? then you can use the css selector to select the inner children

File: src/flipNumbers.js (hunk 1 above)
Comment (zikiro) [reply src/flipNumbers.js]: Do you mean the misnamed prop? If so both `numberStyle` and `nonNumberStyle` are fixed in this pr. If not could you clarify?

Comment (bluebill1049) [reply src/flipNumbers.js]: Hi, just fix this `nonNumberStyle?: React.CSSProperites,`. thanks for the PR 👍

**Comment by bluebill1049** [comment]
sorry i completely miss the pr as well, I got too many notifications.

File: src/flipNumbers.js
Comment (zikiro) [inline src/flipNumbers.js:1]: line 114 of flip number should this be `style={nonNumberStyle}` rather than `className={nonNumberStyle}` ?

**Comment by zikiro** [comment]
> sorry i completely miss the pr as well, I got too many notifications.

No worries!

File: src/flipNumbers.js
Comment (bluebill1049) [reply src/flipNumbers.js:1]: oh yea lol

File: src/flipNumbers.js (hunk 1 above)
Comment (zikiro) [reply src/flipNumbers.js]: Fixed this and one other.

File: src/flipNumbers.js
Comment (zikiro) [reply src/flipNumbers.js:1]: fixed.

**Comment by bluebill1049** [comment]
Just one last point: Could we have one single `className` prop instead `numberClassName` and the other one?

**Comment by zikiro** [comment]
> Just one last point: Could we have one single `className` prop instead `numberClassName` and the other one?

So the point of two class names rather than one is so that people can independently be able to customize around numbers vs non numbers. Without two separate it makes it much harder if you need different surrounding styles. An example of this is if you wanted a box to surround each of the numbers, but not surround the periods/commas.

**Comment by bluebill1049** [comment]
You made a good point there @zikiro

File: src/flipNumbers.js [hunk 2]
```diff
@@ -91,6 +97,7 @@ export default class FlipNumbers extends React.Component<Props> {
                 position={numberCounter}
                 length={numbers.length}
                 activeNumber={parseInt(n, 10)}
+                className={numberContainerClassName}
```
Comment (bluebill1049) [inline src/flipNumbers.js]: maybe just `numberClassName`?

Comment (zikiro) [reply src/flipNumbers.js]: Fixed as requested.

**Review by bluebill1049 — APPROVED** [review]

**Comment by bluebill1049** [comment]
Thanks for the feature 👍

**Comment by zikiro** [comment]
> Thanks for the feature 👍

Thank you for the reviews of this PR!

**Comment by bluebill1049** [comment]
I will release the patch soon, remind me if I forgot 👍

**Comment by zikiro** [comment]
> I will release the patch soon, remind me if I forgot 👍

Will do I'll keep an eye out 👍

**Pull request merged**